tests/test_component/test_states.py

```diff
@@ -1,86 +1,93 @@
-                                         states.SCHEDULED: 4,
-                                         states.SUBMITTING: 5,
-                                         states.COMPLETED: 7,
-                                         states.DONE: 10,
-                                         states.FAILED: 10,
-                                         states.CANCELED: 10
-                                         }
-
-    assert states._task_state_inv == {
-        1: states.INITIAL,
-        2: states.SCHEDULING,
-        4: states.SCHEDULED,
-        5: states.SUBMITTING,
-        7: states.COMPLETED,
-        10: [states.DONE, states.FAILED, states.CANCELED]
-    }
+class TestBase(TestCase):
+
+    def test_states_list(self):
+
+        self.assertEqual(states.INITIAL, 'DESCRIBED')
+        self.assertEqual(states.SCHEDULING, 'SCHEDULING')
+        self.assertEqual(states.SUSPENDED, 'SUSPENDED')
+        self.assertEqual(states.SCHEDULED, 'SCHEDULED')
+        self.assertEqual(states.SUBMITTING, 'SUBMITTING')
+        self.assertEqual(states.COMPLETED, 'EXECUTED')
+        self.assertEqual(states.DONE, 'DONE')
+        self.assertEqual(states.FAILED, 'FAILED')
+        self.assertEqual(states.CANCELED, 'CANCELED')
+        self.assertEqual(states.FINAL, [states.DONE, states.FAILED, states.CANCELED])
+
+
+    def test_state_numeric_vals(self):
+
+        self.assertEqual(states.state_numbers, {states.INITIAL: 1,
+                                        states.SCHEDULING: 2,
+                                        states.SUSPENDED: 3,
+                                        states.SCHEDULED: 4,
+                                        states.SUBMITTING: 5,
+                                        states.COMPLETED: 7,
+                                        states.DONE: 10,
+                                        states.FAILED: 10,
+                                        states.CANCELED: 10
+                                        })
+
+
+    def test_pipeline_states(self):
+
+        self.assertEqual(states._pipeline_state_values, {states.INITIAL: 1,
+                                                states.SCHEDULING: 2,
+                                                states.SUSPENDED: 3,
+                                                states.DONE: 10,
+                                                states.FAILED: 10,
+                                                states.CANCELED: 10
+                                                })
+
+        self.assertEqual(states._pipeline_state_inv, {1: states.INITIAL,
+                                            2: states.SCHEDULING,
+                                            3: states.SUSPENDED,
+                                            10: [states.DONE, states.FAILED,
+                                                states.CANCELED]
+                                            })
+
+
+    def test_stage_states(self):
+
+        self.assertEqual(states._stage_state_values, {states.INITIAL: 1,
```

Comment:
Why is `assertEqual(a, literal)` better / how is it different from `assert(a == literal)`?

Reply:
The assert commands are ignored when python is executed with the -O flag as they are meant to be used mainly for debugging. See [here](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement). Also, the methods offered by `TestCase` are supposed to provide better messages when a test fails ([here](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual)).

Does this help?

Reply:
Not sure if we would ever run tests with `-O`, but the better message makes sense, thanks.